http-common/src/connector.rs

```diff
@@ -768,8 +754,56 @@ fn get_num_systemd_sockets() -> Result<Option<std::os::unix::io::RawFd>, String>
         }
     }
 
-    // Return the number of systemd sockets.
-    Ok(Some(listen_fds))
+    //If there is only one socket, we know this is the identity service which uses only one socket, so we have a match:
+    if listen_fds == 1 {
+        return Ok(Some(SD_LISTEN_FDS_START as i32));
```

Comment:
Do you need the `as i32` here? If so, it would be better to have it `as std::os::unix::io::RawFd` to match the return type of this function.

Reply:
You are right